timeseries/src/autogluon/timeseries/models/chronos/model.py

```diff
@@ -310,23 +312,38 @@ def _predict(
             )
             self.model_pipeline.model.eval()
             with torch.inference_mode():
-                prediction_samples = [
-                    self.model_pipeline.predict(
-                        batch,
-                        prediction_length=self.prediction_length,
-                        num_samples=self.num_samples,
-                        limit_prediction_length=False,
-                    )
-                    .detach()
-                    .cpu()
-                    .numpy()
-                    for batch in inference_data_loader
-                ]
-
-        samples = np.concatenate(prediction_samples, axis=0).swapaxes(1, 2).reshape(-1, self.num_samples)
-
-        mean = samples.mean(axis=-1, keepdims=True)
-        quantiles = np.quantile(samples, self.quantile_levels, axis=-1).T
+                if self.model_pipeline.forecast_type == ForecastType.SAMPLES:
```

Comment:
With this design we are implicitly making a decision that I'd like to discuss to make sure it's the right one. `ChronosModel` will be a generic wrapper for HF models that are loaded with `from_pretrained` and implement the `predict(context: torch.Tensor, prediction_length: int) -> torch.Tensor` API. To control which model is used, the users need to specify the `model_path` hyperparameter.
```python
hyperparameters={"Chronos": {"model_path": "autogluon/chronos-bolt-small"}}
```

The alternative option would involve creating a separate class (e.g., `ChronosBoltModel`) for the new set of Chronos models that directly return the quantile forecasts. This could be slightly more intuitive for the users since one hyperparameter completely changes the model architecture
```python
hyperparameters={"ChronosBolt": {"model_path": "small"}}
```

The first option seems cleaner and more future proof to me, but I anyway wanted to bring this up to make sure that we don't miss any important limitations of the chosen design.

Reply:
I am also comfortable with (1) for now. A "ChronosModel" in this sense is anything that can be loaded by a `ForecastPipeline` (which I think we should also rename to something like `BaseChronosPipeline`). That means it is compatible with our Pipeline API, as well as persisted with a Chronos config or `chronos_pipeline_class`. We could alias to get the effect of (2), though I'm a bit afraid that could complicate things.